core/src/test/java/org/apache/calcite/materialize/NormalizationTrimFieldTest.java

```diff
@@ -106,4 +116,41 @@
     final String relOptimizedStr = RelOptUtil.toString(relOptimized.get(0).getKey());
     assertThat(isLinux(optimized).matches(relOptimizedStr), is(true));
   }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-4391">[CALCITE-4391]
+   * The order of join conditions is different,materialized view recognition fails</a>. */
+  @Test void testJoinToJoinConditionReorder() {
+    final RelBuilder relBuilder = RelBuilder.create(config().build());
```

Comment:
The order of join conditions is different, materialized view recognition fails -> materialized view recognition fails when the order of join conditions is different?